### ISSUE TITLE
Pkg.submit() unable to get git credentials

### ISSUE THREAD

**Issue by ViralBShah** [JuliaLang/julia#13528]
In trying to fix some deprecation warnings, I figured I will try `Pkg.submit` as it takes care of a lot of stuff. It kept asking me for my github password, and finally gave up.

````
julia> Pkg.submit("RDatasets")
INFO: Forking johnmyleswhite/RDatasets.jl to ViralBShah
Enter host password for user 'ViralBShah':
INFO: Retrieving existing GitHub token. (You may have to re-enter your password.)
Enter host password for user 'ViralBShah':
INFO: Could not authenticate with existing token. Deleting token and trying again.
Enter host password for user 'ViralBShah':
INFO: Retrieving existing GitHub token. (You may have to re-enter your password.)
Enter host password for user 'ViralBShah':
ERROR: forking johnmyleswhite/RDatasets.jl failed: Bad credentials
 in error at /Users/viral/Desktop/Julia-0.4.0.app/Contents/Resources/julia/lib/julia/sys.dylib
 in fork at pkg/github.jl:144
 in pull_request at pkg/entry.jl:327
 in submit at pkg/entry.jl:339
 in submit at pkg/entry.jl:337
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
 in submit at pkg.jl:58
````

**Comment by tkelman** [comment]
Might be https://github.com/JuliaLang/julia/pull/12607 ? That was merged to master a little late to get it into 0.4.0 final I think. Could you try with that patch locally on a source build of release-0.4, or see if master works any better? Though there are lots of libgit2-specific issues open at the moment.

**Comment by wildart** [comment]
Yes, #12607 should have addressed this issue. If this happens again, reopen the issue in [PkgDev](https://github.com/JuliaLang/PkgDev.jl) because `submit` was moved there.
